### ISSUE TITLE
examples: Create a pions to pions example

### ISSUE THREAD

**Issue by backkem** [pions/webrtc#40]
Right now we only have examples of pions connecting to a browser. It would be nice to have an example of two pions clients connecting to each other.

**Comment by Sean-Der** [comment]
This is blocked by https://github.com/pions/webrtc/issues/23

Once we land this will be a great first issue, but will mark blocked for now

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 2 months, 4 weeks from now.
Please review their action plans below:


1. [writeprovidence](https://gitcoin.co/profile/writeprovidence) has started work. 
    
    will be very much interested in working on this project, i will always keep intouch if am lost.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/pions/webrtc/40/869).

**Comment by backkem** [comment]
Welcome @writeprovidence, glad to have you. We're very active on the #pions channel of the golang slack if you have any questions. Please note that this issue is still blocked by #23 for the moment.

**Comment by gitcoinbot** [comment]
@writeprovidence Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=1) | [3 days](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=3) | [5 days](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=5) | [10 days](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=10) | [100 days](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=100)

**Comment by gitcoinbot** [comment]
@writeprovidence Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=1) | [3 days](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=3) | [5 days](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=5) | [10 days](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=10) | [100 days](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=100)

**Comment by writeprovidence** [comment]
yes

**Comment by writeprovidence** [comment]
hello @backkem pls could you give a little more explanation on this, thanks .

**Comment by backkem** [comment]
Currently all our examples demonstrate connections between a pions client and a web browser. This issue is meant to introduce one (or more) example(s) that demonstrate a connection between two pions (Go) clients. I suggest you start by looking at the existing examples. For example: gstreamer-send and gstreamer-receive (or save-to-disk). These could be combined into one example so one pions client sends media and another pions client receives the media. Another option would be to make an example similar to the data-channels example where one pions client sends data over a data channel and another pions client receives the data.
As mentioned, this issue is currently blocked by #23 and #24 (that we know off). However, we're actively working on getting those fixed. In the mean time, the API for creating the example(s) already exists and should be stable. This means you could start creating the example. Unfortunately, you'll have to wait until the blocking issues are fixed in order to test them.

**Comment by gitcoinbot** [comment]
@writeprovidence Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=1) | [3 days](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=3) | [5 days](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=5) | [10 days](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=10) | [100 days](https://gitcoin.co/issue/pions/webrtc/40/869?snooze=100)

**Comment by writeprovidence** [comment]
[Client connection.pdf](https://github.com/pions/webrtc/files/2278236/Client.connection.pdf)

**Comment by writeprovidence** [comment]
hello@backem pls review the code if am on the right track. its on pdf
waiting for ur reply.

**Comment by backkem** [comment]
I'm sorry @writeprovidence but we don't accept contributions via PDF. I've deleted the related comment for security reasons. If you want to contribute, you'll have to send in a pull request with an example, written in Golang, similar to the current examples in the examples directory.

I fear this issue may be outside of your current abilities. It requires you to possess working knowledge of Golang, Git, JavaScript and knowledge of WebRTC. May I kindly ask to release (called "stop work" on GitCoin) this issue of you agree.

**Comment by writeprovidence** [comment]
ok. thanks for ur patience. God bless

**Comment by Sean-Der** [comment]
I moved this to our [Project Ideas](https://github.com/pions/webrtc/wiki/Contributing#project-ideas)

Feel free to reopen this when this is active again!

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Cancelled** 

<hr>

__The funding of 0.1 ETH  (25.5 USD @ $255.01/ETH) attached to this issue has been **cancelled** by the bounty submitter__

 
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $18,118.10 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
